src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Cipher.java

```diff
@@ -628,7 +635,7 @@ private int implUpdate(byte[] in, int inOfs, int inLen,
                 throw (ShortBufferException)
                         (new ShortBufferException().initCause(e));
             }
-            reset(false);
+            reset(true);
```

Comment:
Per PKCS#11 spec, "A call to C_EncryptUpdate which results in an error other than CKR_BUFFER_TOO_SMALL terminates the current encryption operation.", so I'd expect comment here to explain why we are doing reset(true) here. If not mentioning the known NSS behavior which triggered this change, at least comment the bug id so we don't lost track of the reason for the switch.

Reply:
Yes, makes sense. Thanks